File: src/game/entity/enemies/GhostEnemy.java
package game.entity.enemies;

import game.entity.enemies.npcs.IntelligentEnemy;
import game.entity.models.Entity;
import game.models.Direction;
import game.ui.panels.game.PitchPanel;
import game.utils.Paths;

import java.util.HashSet;
import java.util.Set;

public class GhostEnemy extends IntelligentEnemy {
    public GhostEnemy(){
        hitboxSizetoWidthRatio = 0.837f;
        hitboxSizeToHeightRatio= 1;
        imagePossibleDirections.remove(Direction.UP);
        imagePossibleDirections.remove(Direction.DOWN);
    }
    @Override
    public String[] getCharacterOrientedImages() {
        return new String[]{String.format("%s/mini_ghost/ghost_%s.png", Paths.getEnemiesFolder(), imageDirection.toString().toLowerCase())};

    }

    @Override
    public Set<Class<? extends Entity>> getObstacles() {
        return new HashSet<>();
    }
    @Override
    public int getSize() {
        return PitchPanel.COMMON_DIVISOR * 2;
    }
}

File: src/game/engine/world/domain/entity/actors/impl/enemies/npcs/GhostEnemy.kt
package game.engine.world.domain.entity.actors.impl.enemies.npcs

import game.engine.world.types.EntityTypes
import game.engine.world.domain.entity.geo.Direction
import game.engine.world.domain.entity.actors.abstracts.base.Entity
import game.engine.ui.panels.game.PitchPanel
import game.utils.Paths.enemiesFolder

class GhostEnemy : IntelligentEnemy {
    init {
        hitboxSizeToWidthRatio = 0.837f
        hitboxSizeToHeightRatio = 1f
    }

    constructor() : super()
    constructor(id: Long) : super(id)

    override fun getCharacterOrientedImages(): Array<String> =
            arrayOf("$enemiesFolder/mini_ghost/ghost_${imageDirection.toString().lowercase()}.png")

    override fun getImageDirections(): List<Direction> = listOf(Direction.RIGHT, Direction.LEFT)

    override fun getObstacles(): Set<Class<out Entity?>> = interactionsEntities

    override val size: Int
        get() = PitchPanel.COMMON_DIVISOR * 2

    override val type: EntityTypes
        get() = EntityTypes.GhostEnemy
}